core/dbt/clients/yaml_helper.py

```diff
@@ -68,13 +69,16 @@ def contextualized_yaml_error(raw_contents, error):
     )
 
 
-def safe_load(contents) -> Optional[Dict[str, Any]]:
-    return yaml.load(contents, Loader=UniqueKeyLoader)
+def safe_load(contents, unique=False) -> Optional[Dict[str, Any]]:
+    if unique:
+        return yaml.load(contents, Loader=UniqueKeyLoader)
+    else:
+        return yaml.load(contents, Loader=SafeLoader)
```

Comment:
Question for @dbt-labs/core-language on how to factor this code: Should we opt for one method with a boolean argument like this, or for two different methods: `load_and_validate_unique_keys` + `safe_load`?

Reply:
Whether I prefer a 'unique' argument or a separate method name depends on why we're only actually using the 'unique' arg in 'load_yaml_text', but we call 'safe_load' in multiple other places. Do those other places not need the duplicate key validation? Why is it only called there?

Reply:
That's a good point - I think we should be validating duplicate key everywhere.. so I reverted this to what it was before.